diff --git a/Configurations/Common.xcconfig b/Configurations/Common.xcconfig
index 1c63d37..3b7ebd2 100644
--- a/Configurations/Common.xcconfig
+++ b/Configurations/Common.xcconfig
@@ -33,6 +33,6 @@ THIRD_PARTY_LIB_PATH[sdk=iphonesimulator*] = $(PROJECT_DIR)/libs/dependency/ioss
 THIRD_PARTY_LIB_PATH[sdk=xros*] = $(PROJECT_DIR)/libs/dependency/visionos
 THIRD_PARTY_LIB_PATH[sdk=xrsimulator*] = $(PROJECT_DIR)/libs/dependency/visionossim
 
-SYSTEM_HEADER_SEARCH_PATHS = $(PROJECT_DIR) $(PROJECT_DIR)/../Celestia/src $(THIRD_PARTY_LIB_PATH)/include $(THIRD_PARTY_LIB_PATH)/include/miniaudio $(THIRD_PARTY_LIB_PATH)/include/meshoptimizer $(THIRD_PARTY_LIB_PATH)/include/eigen3 $(THIRD_PARTY_LIB_PATH)/include/luajit $(THIRD_PARTY_LIB_PATH)/include/gettext $(THIRD_PARTY_LIB_PATH)/include/jpeg $(THIRD_PARTY_LIB_PATH)/include/libpng $(THIRD_PARTY_LIB_PATH)/include/cspice $(THIRD_PARTY_LIB_PATH)/include/freetype/freetype2 $(THIRD_PARTY_LIB_PATH)/include/icu $(THIRD_PARTY_LIB_PATH)/include/libepoxy  $(THIRD_PARTY_LIB_PATH)/include/boost
+SYSTEM_HEADER_SEARCH_PATHS = $(PROJECT_DIR) $(PROJECT_DIR)/../Celestia/src $(THIRD_PARTY_LIB_PATH)/include $(THIRD_PARTY_LIB_PATH)/include/miniaudio $(THIRD_PARTY_LIB_PATH)/include/meshoptimizer $(THIRD_PARTY_LIB_PATH)/include/eigen3 $(THIRD_PARTY_LIB_PATH)/include/luajit $(THIRD_PARTY_LIB_PATH)/include/gettext $(THIRD_PARTY_LIB_PATH)/include/jpeg $(THIRD_PARTY_LIB_PATH)/include/libpng $(THIRD_PARTY_LIB_PATH)/include/cspice $(THIRD_PARTY_LIB_PATH)/include/freetype/freetype2 $(THIRD_PARTY_LIB_PATH)/include/icu $(THIRD_PARTY_LIB_PATH)/include/angle  $(THIRD_PARTY_LIB_PATH)/include/boost
 
 STRIP_INSTALLED_PRODUCT = NO
diff --git a/Configurations/Debug.xcconfig b/Configurations/Debug.xcconfig
index 3f921cc..2247fff 100644
--- a/Configurations/Debug.xcconfig
+++ b/Configurations/Debug.xcconfig
@@ -11,9 +11,6 @@
 
 #include "Common.xcconfig"
 
-PREPROCESSOR_BASE = DEBUG=1 ENABLE_NLS USE_SPICE USE_MINIAUDIO CELX USE_ICU BOOST_NO_EXCEPTIONS
+PREPROCESSOR_BASE = DEBUG=1 ENABLE_NLS USE_SPICE USE_MINIAUDIO CELX USE_ICU BOOST_NO_EXCEPTIONS GL_ES
 
 GCC_PREPROCESSOR_DEFINITIONS = $(PREPROCESSOR_BASE)
-GCC_PREPROCESSOR_DEFINITIONS[sdk=macosx*] = $(PREPROCESSOR_BASE)
-GCC_PREPROCESSOR_DEFINITIONS[sdk=iphoneos*] = $(PREPROCESSOR_BASE) GL_ES
-GCC_PREPROCESSOR_DEFINITIONS[sdk=iphonesimulator*] = $(PREPROCESSOR_BASE) GL_ES
diff --git a/Configurations/Release.xcconfig b/Configurations/Release.xcconfig
index 338ab27..5ef5024 100644
--- a/Configurations/Release.xcconfig
+++ b/Configurations/Release.xcconfig
@@ -11,9 +11,6 @@
 
 #include "Common.xcconfig"
 
-PREPROCESSOR_BASE = NDEBUG NO_DEBUG EIGEN_NO_DEBUG ENABLE_NLS USE_SPICE USE_MINIAUDIO CELX USE_ICU BOOST_NO_EXCEPTIONS
+PREPROCESSOR_BASE = NDEBUG NO_DEBUG EIGEN_NO_DEBUG ENABLE_NLS USE_SPICE USE_MINIAUDIO CELX USE_ICU BOOST_NO_EXCEPTIONS GL_ES
 
 GCC_PREPROCESSOR_DEFINITIONS = $(PREPROCESSOR_BASE)
-GCC_PREPROCESSOR_DEFINITIONS[sdk=macosx*] = $(PREPROCESSOR_BASE)
-GCC_PREPROCESSOR_DEFINITIONS[sdk=iphoneos*] = $(PREPROCESSOR_BASE) GL_ES
-GCC_PREPROCESSOR_DEFINITIONS[sdk=iphonesimulator*] = $(PREPROCESSOR_BASE) GL_ES
