diff --git a/MobileCelestia.xcodeproj/project.pbxproj b/MobileCelestia.xcodeproj/project.pbxproj
index 33b4a13..fbd1a17 100644
--- a/MobileCelestia.xcodeproj/project.pbxproj
+++ b/MobileCelestia.xcodeproj/project.pbxproj
@@ -1730,7 +1730,6 @@
 					"-ljpeg",
 					"-lfreetype",
 					"-lintl",
-					"-lGL",
 					"-lzip",
 					"$(THIRD_PARTY_LIB_PATH)/libs/cspice.a",
 				);
@@ -1781,7 +1780,6 @@
 					"-ljpeg",
 					"-lfreetype",
 					"-lintl",
-					"-lGL",
 					"-lzip",
 					"$(THIRD_PARTY_LIB_PATH)/libs/cspice.a",
 				);
diff --git a/MobileCelestia/Celestia/CelestiaDisplayController.swift b/MobileCelestia/Celestia/CelestiaDisplayController.swift
index 94b2dcf..3d630a8 100644
--- a/MobileCelestia/Celestia/CelestiaDisplayController.swift
+++ b/MobileCelestia/Celestia/CelestiaDisplayController.swift
@@ -60,7 +60,7 @@ class CelestiaDisplayController: AsyncGLViewController {
 
     init(msaaEnabled: Bool, screen: UIScreen, initialFrameRate frameRate: Int, executor: AsyncGLExecutor, subscriptionManager: SubscriptionManager) {
 #if targetEnvironment(macCatalyst)
-        let api = AsyncGLAPI.openGLLegacy
+        let api = AsyncGLAPI.openGLES2
 #else
         let api = AsyncGLAPI.openGLES2
 #endif
diff --git a/Podfile b/Podfile
index 0ec4233..1268273 100644
--- a/Podfile
+++ b/Podfile
@@ -11,7 +11,7 @@ target 'MobileCelestia' do
   pod 'AppCenter/Crashes', appcenter_version
 
   pod "MWRequest", :git => "https://github.com/levinli303/mwrequest.git", :tag => mwrequest_version_tag
-  pod "AsyncGL/OpenGL", :git => "https://github.com/levinli303/AsyncGL.git", :tag => asyncgl_version_tag
+  pod "AsyncGL/ANGLE", :git => "https://github.com/levinli303/AsyncGL.git", :tag => asyncgl_version_tag
 end
 
 target 'CelestiaUI' do
