diff --git a/MobileCelestia.xcodeproj/project.pbxproj b/MobileCelestia.xcodeproj/project.pbxproj
index 465e11e..afeef0b 100644
--- a/MobileCelestia.xcodeproj/project.pbxproj
+++ b/MobileCelestia.xcodeproj/project.pbxproj
@@ -1713,7 +1713,6 @@
 					"-ljpeg",
 					"-lfreetype",
 					"-lintl",
-					"-lGL",
 					"$(THIRD_PARTY_LIB_PATH)/libs/cspice.a",
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = space.celestia.MobileCelestia;
@@ -1763,7 +1762,6 @@
 					"-ljpeg",
 					"-lfreetype",
 					"-lintl",
-					"-lGL",
 					"$(THIRD_PARTY_LIB_PATH)/libs/cspice.a",
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = space.celestia.MobileCelestia;
diff --git a/MobileCelestia/Celestia/CelestiaDisplayController.swift b/MobileCelestia/Celestia/CelestiaDisplayController.swift
index aebe510..3632e64 100644
--- a/MobileCelestia/Celestia/CelestiaDisplayController.swift
+++ b/MobileCelestia/Celestia/CelestiaDisplayController.swift
@@ -60,7 +60,7 @@ class CelestiaDisplayController: AsyncGLViewController {
 
     init(msaaEnabled: Bool, screen: UIScreen, initialFrameRate frameRate: Int, executor: AsyncGLExecutor, subscriptionManager: SubscriptionManager) {
 #if targetEnvironment(macCatalyst)
-        let api = AsyncGLAPI.openGLLegacy
+        let api = AsyncGLAPI.openGLES2
 #else
         let api = AsyncGLAPI.openGLES2
 #endif
diff --git a/Podfile b/Podfile
index 5652a8d..027daa9 100644
--- a/Podfile
+++ b/Podfile
@@ -13,7 +13,7 @@ target 'MobileCelestia' do
   pod 'ZIPFoundation', zipfoundation_version
 
   pod "MWRequest", :git => "https://github.com/levinli303/mwrequest.git", :tag => mwrequest_version_tag
-  pod "AsyncGL/OpenGL", :git => "https://github.com/levinli303/AsyncGL.git", :tag => asyncgl_version_tag
+  pod "AsyncGL/ANGLE", :git => "https://github.com/levinli303/AsyncGL.git", :tag => asyncgl_version_tag
 end
 
 target 'CelestiaUI' do
