diff --git a/MobileCelestia.xcodeproj/project.pbxproj b/MobileCelestia.xcodeproj/project.pbxproj
index 3006bfb..96b7e7e 100644
--- a/MobileCelestia.xcodeproj/project.pbxproj
+++ b/MobileCelestia.xcodeproj/project.pbxproj
@@ -1298,6 +1298,9 @@
 				kind = exactVersion;
 				version = 1.2.0;
 			};
+			traits = (
+				ANGLE,
+			);
 		};
 		EAC20FDF2CA0CB7C0098AABA /* XCRemoteSwiftPackageReference "sentry-cocoa" */ = {
 			isa = XCRemoteSwiftPackageReference;
diff --git a/MobileCelestia/Celestia/CelestiaDisplayController.swift b/MobileCelestia/Celestia/CelestiaDisplayController.swift
index 99bcf2a..5583db6 100644
--- a/MobileCelestia/Celestia/CelestiaDisplayController.swift
+++ b/MobileCelestia/Celestia/CelestiaDisplayController.swift
@@ -68,7 +68,7 @@ class CelestiaDisplayController: AsyncGLViewController {
         featureFlags: FeatureFlags
     ) {
 #if targetEnvironment(macCatalyst)
-        let api = AsyncGLAPI.openGLLegacy
+        let api = AsyncGLAPI.openGLES2
 #else
         let api = featureFlags.iosGLESV3 ? AsyncGLAPI.openGLES3 : AsyncGLAPI.openGLES2
 #endif
