diff --git a/MobileCelestia.xcodeproj/project.pbxproj b/MobileCelestia.xcodeproj/project.pbxproj
index 3d11a5b..6164de9 100644
--- a/MobileCelestia.xcodeproj/project.pbxproj
+++ b/MobileCelestia.xcodeproj/project.pbxproj
@@ -1423,6 +1423,9 @@
 				kind = exactVersion;
 				version = 1.2.1;
 			};
+			traits = (
+				ANGLE,
+			);
 		};
 		EAC20FDF2CA0CB7C0098AABA /* XCRemoteSwiftPackageReference "sentry-cocoa" */ = {
 			isa = XCRemoteSwiftPackageReference;
diff --git a/MobileCelestia/Celestia/CelestiaDisplayController.swift b/MobileCelestia/Celestia/CelestiaDisplayController.swift
index 288df91..bfdfad8 100644
--- a/MobileCelestia/Celestia/CelestiaDisplayController.swift
+++ b/MobileCelestia/Celestia/CelestiaDisplayController.swift
@@ -68,7 +68,7 @@ class CelestiaDisplayController: AsyncGLViewController {
         featureFlags: FeatureFlags
     ) {
 #if targetEnvironment(macCatalyst)
-        let api = AsyncGLAPI.openGLCore41
+        let api = AsyncGLAPI.openGLES3
 #else
         let api = AsyncGLAPI.openGLES3
 #endif
